src/rgw/rgw_sal.h

```diff
@@ -1667,6 +1669,55 @@ class Lifecycle {
 						       const std::string& cookie) = 0;
 };
 
+/** @brief Abstraction of a serializer for Restore
+ */
+class RestoreSerializer : public Serializer {
+public:
+  RestoreSerializer() {}
+  virtual ~RestoreSerializer() = default;
+};
+
+/**
+ * @brief Abstraction for restore processing
+ *
+ * The Restore class is designed to manage the restoration of objects
+ * from cloud tier storage back into the Ceph cluster. This is particularly used
+ * for objects stored in cold storage solutions like AWS Glacier or Tape-based systems,
+ * where retrieval operations are asynchronous and can take a significant amount of time.
+ */
+class Restore {
+
+public:
+  Restore() = default;
+  virtual ~Restore() = default;
+  virtual int initialize(const DoutPrefixProvider* dpp, optional_yield y,
+		  int n_objs, std::vector<std::string>& obj_names) = 0;  
+  /** Add a single restore entry state */
+  virtual int add_entry(const DoutPrefixProvider* dpp, optional_yield y,
```

Comment:
You don't really need this to be separate. The only reason DataLog has them separate is because of the abstraction between OMAP and FIFO. `add_entry` is a just a special case of `add_entries`

Reply:
Done.